test/opset1.cpp

```diff
@@ -0,0 +1,141 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "gtest/gtest.h"
+
+#include "ngraph/ngraph.hpp"
+#include "ngraph/opsets/opset1.hpp"
+
+#include <memory>
+using namespace std;
+using namespace ngraph;
+
+#define CHECK_OPSET(op1, op2) EXPECT_TRUE(is_type<op1>(make_shared<op2>()));
```

Comment:
Just wondering if instead of testing it in runtime, we could switch to `std::is_same` and have it checked in compile time.

https://ideone.com/WYkqla something like this

Reply:
We could possibly do both. We are checking the 'is_type' implementation as well in this test.